src/decays/H_SM_decays/decay_H_to_ZZ.cpp

```diff
@@ -0,0 +1,126 @@
+// special case for Higgs -> Z Z
+// TODO: implement higher order corrections
+
+struct my_f_params {double mHOS; double mVOS; double GammaV;};
+
+double hVV_4body(double *q2, size_t dim, void *params)
+{
+  (void)(dim); /* avoid unused parameter warnings */
+  struct my_f_params * fp = (struct my_f_params *)params;
+  const double mHOS = fp->mHOS;
+  if (q2[1] > Sqr(mHOS - std::sqrt(q2[0]))) return 0.;
+  const double mVOS = fp->mVOS;
+  const double GammaV = fp->GammaV;
+  const double kl = KallenLambda(1., q2[0]/Sqr(mHOS), q2[1]/Sqr(mHOS));
+  return
+     mVOS*GammaV/(Sqr(q2[0] - Sqr(mVOS)) + Sqr(mVOS*GammaV))
+     * mVOS*GammaV/(Sqr(q2[1] - Sqr(mVOS)) + Sqr(mVOS*GammaV))
+     * std::sqrt(kl)*(kl + 12.*q2[0]*q2[1]/Power4(mHOS));
+}
+
+template <>
+double CLASSNAME::get_partial_width<H,Z,Z>(
+   const context_base& context,
+   typename field_indices<H>::type const& indexIn,
+   typename field_indices<Z>::type const& indexOut1,
+   typename field_indices<Z>::type const& indexOut2
+   ) const
+{
+
+   const double mHOS = context.physical_mass<H>(indexIn);
+   // There might be large differences between mZ from mass block
+   // and one from slha input, especially in the decoupling limit
+   // so we use the latter one. There might be a problem with
+   // models where Z mixes with something else.
+   // const double mZOS = context.physical_mass<Z>(indexOut1);
+   const double mZOS = qedqcd.displayPoleMZ();
+   const double x = Sqr(mZOS/mHOS);
+   double res = 0;
+
+   // mH < mZ
+   if ((x > 1.0 && flexibledecay_settings.get(FlexibleDecay_settings::offshell_VV_decays) != 0) ||
+       (4.*x > 1.0 && flexibledecay_settings.get(FlexibleDecay_settings::offshell_VV_decays) == 2)) {
+
+      // integrand
+      constexpr double GammaZ = 2.4952;
+      struct my_f_params params = {mHOS, mZOS, GammaZ};
+      gsl_monte_function G = {&hVV_4body, 2, &params};
+
+      // setup integration
+      gsl_rng_env_setup ();
+      double xl[2] = {0, 0};
+      double xu[2] = {Sqr(mHOS), Sqr(mHOS)};
+      // this gives relative error < 0.05%
+      constexpr size_t calls = 1'000'000;
+      double err;
+      const gsl_rng_type *T = gsl_rng_default;
+      gsl_rng *r = gsl_rng_alloc (T);
+      gsl_monte_miser_state *s = gsl_monte_miser_alloc (2);
+      gsl_monte_miser_integrate (&G, xl, xu, 2, calls, r, s,
+                                 &res, &err);
+
+      // clean-up
+      gsl_monte_miser_free (s);
+      gsl_rng_free (r);
+
+      // prefactor
+      const auto indices = concatenate(indexIn, indexOut2, indexOut1);
+      const auto ghZZ =
+         Vertex<H, Z, Z>::evaluate(indices, context).value();
+      res *= 1./(16.*Cube(Pi))* std::norm(ghZZ) * 0.25*Cube(mHOS)/Power4(mZOS)/2.;
+      err *= 1./(16.*Cube(Pi))* std::norm(ghZZ) * 0.25*Cube(mHOS)/Power4(mZOS)/2.;
```

Comment:
Expression appears twice. Better only calculate once.

Reply:
fixed in 0639a17a3d2b914b83db6771d5401437c6673002

Reply:
Thanks a lot!